core-bundle/src/EventListener/DataContainer/DefaultOperationsListener.php

```diff
@@ -227,6 +227,15 @@ private function copyAction(string $table, DataContainerOperation $operation): C
         unset($new['id']);
         $new['tstamp'] = 0;
 
+        // Unset the PID field for the copy operation (act=paste&mode=copy), so a voter can differentiate
+        // between the copy operation (without PID) and the paste button (with new target PID)
+        if (
+            ($GLOBALS['TL_DCA'][$table]['list']['sorting']['mode'] ?? null) === DataContainer::MODE_TREE
+            || !empty($GLOBALS['TL_DCA'][$table]['config']['ptable'] ?? null)
```

Comment:
```suggestion
            || !empty($GLOBALS['TL_DCA'][$table]['config']['ptable'])
```

`?? null` should not be necessary inside `empty()`.